src/com/github/alex_moon/porcupi/manager/ManagerClient.java

```diff
@@ -35,9 +56,24 @@ public ManagerClient() {
             System.exit(2);
         }
     }
-    
-    public List<String> manage(String inputLine) {
-        System.out.println("received from server: " + inputLine);
-        return null;
+
+    public void tellIn(Message input) {
+        if (input.getContext() != null) {
+            if (!input.getContext().equals(context)) {
+                System.out.println("now operating in context " + input.getContext());
+            }
+        } else {
+            if (context != null) {
+                System.out.println("closing context " + context);
+            }
+        }
+        context = input.getContext();
```

Comment:
not convinced this is correct - why would we want to close the context on a server message unrelated to that context?